pkg/network/downwardapi/networkinfo_test.go

```diff
@@ -46,33 +46,46 @@ var _ = Describe("Network info", func() {
 		Expect(err).ToNot(HaveOccurred())
 		Expect(networkInfo.Interfaces).To(ConsistOf(expectedInterfaces))
 	})
+
 	It("should create an empty network info annotation value when there are no networks", func() {
 		networkDeviceInfoMap := map[string]*networkv1.DeviceInfo{}
 
 		Expect(downwardapi.CreateNetworkInfoAnnotationValue(networkDeviceInfoMap)).To(Equal("{}"))
 	})
-	It("should produce a deterministic and ordered output regardless of the map key order", func() {
+
+	It("should produce a deterministic and output sorted by network name regardless of the map key order", func() {
 		deviceInfo1 := &networkv1.DeviceInfo{Type: "type1"}
 		deviceInfo2 := &networkv1.DeviceInfo{Type: "type2"}
-		map1 := map[string]*networkv1.DeviceInfo{
-			"netB": deviceInfo1,
-			"netA": deviceInfo2,
-		}
-		map2 := map[string]*networkv1.DeviceInfo{
-			"netA": deviceInfo2,
-			"netB": deviceInfo1,
+		deviceInfo3 := &networkv1.DeviceInfo{Type: "type3"}
+
+		deviceInfoByNetName1 := map[string]*networkv1.DeviceInfo{
+			"netA": deviceInfo1,
+			"netB": deviceInfo2,
+			"netC": deviceInfo3,
 		}
 
-		annotation1 := downwardapi.CreateNetworkInfoAnnotationValue(map1)
-		annotation2 := downwardapi.CreateNetworkInfoAnnotationValue(map2)
+		deviceInfoByNetName2 := map[string]*networkv1.DeviceInfo{
+			"netC": deviceInfo3,
+			"netB": deviceInfo2,
+			"netA": deviceInfo1,
+		}
 
-		Expect(annotation1).To(Equal(annotation2))
+		annotationValue1 := downwardapi.CreateNetworkInfoAnnotationValue(deviceInfoByNetName1)
+		annotationValue2 := downwardapi.CreateNetworkInfoAnnotationValue(deviceInfoByNetName2)
+		Expect(annotationValue1).To(Equal(annotationValue2))
 
-		var networkInfo downwardapi.NetworkInfo
-		err := json.Unmarshal([]byte(annotation1), &networkInfo)
+		var actualNetworkInfo downwardapi.NetworkInfo
+		err := json.Unmarshal([]byte(annotationValue2), &actualNetworkInfo)
 		Expect(err).ToNot(HaveOccurred())
```

Comment:
nit: `Expect(json.Unmarshal([]byte(annotationValue2), &actualNetworkInfo)).To(Succeed())`